plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/connection/ConnectionsRegistry.java

```diff
@@ -150,54 +142,37 @@ public boolean remove(IConnection connection) {
 		}
 	}
 
-	private void fireChange(IConnection  connection, int event) {
+	private void fireChange(IConnection connection, int event) {
 		fireChange(connection, event, null, null, null);
 	}
-	private void fireChange(IConnection  connection, int event, String property, Object oldValue, Object newValue) {
+
+	private void fireChange(IConnection connection, int event, String property, Object oldValue, Object newValue) {
 		if (connection == null) {
 			return;
 		}
-		Job job = new AbstractDelegatingMonitorJob("Connection Changed") {
-		
-			@Override
-			protected IStatus doRun(IProgressMonitor monitor) {
-				try {
-					Iterator<IConnectionsRegistryListener> i = getListeners().iterator();
-					while (i.hasNext()) {
-						IConnectionsRegistryListener l = i.next();
-						switch (event) {
-						case ADDED:
-							l.connectionAdded(connection);
-							break;
-						case REMOVED:
-							l.connectionRemoved(connection);
-							break;
-						case CHANGED:
-							l.connectionChanged(connection, property, oldValue, newValue);
-							break;
-							
-						default:
-							break;
-						}
-					}
-				}catch(Exception e) {
-					return new Status(IStatus.ERROR,OpenShiftCommonCoreActivator.PLUGIN_ID,"Exception while notifying of a connection change",e);
-				}	
-				return Status.OK_STATUS;
+
```

Comment:
I believe that the basic observation with keeping the update sequence intact is a very valid point - using jobs wont allow this since there's no assurance on when jobs are schedules and they can also get reordered in a pool.
The objection with possible UI freezes is also valid, but then ensuring no freeze happens (aka use jobs) should be pushed to the reciever of the notification I believe